### DISCUSSION TITLE
Snow

### DISCUSSION THREAD

**Discussion by Aivean** [CleverRaven/Cataclysm-DDA#46328] [Ideas]
I was considering the idea of adding snow to the game.
 
Found this PR: #28227, it adds "deep snow" as a furniture, which can be problematic from the implementation and performance standpoint.

I suggest a much simpler approach:
* Have a bit matrix the size of the map, where "1" means "terrain is covered in snow"
* During the precipitation, if the temperature is below freezing, an "outside" tile has the random chance to have it's snow bit set to "1"
* When temperature is above freezing, reverse effect happens: snow bits have random chance to disappear

Tileset support: 
* "snow covered" terrain will have it's own prefix that replaces seasonal prefix. 
     E.g. there will be `t_dirt`, `t_dirt_season_winter` and `t_dirt_snow_covered`
* If the "snow covered" tile variant doesn't exist, it falls back to seasonal variant which falls back to generic variant

This will simulate "shallow snow", mostly for cosmetic purposes, although in the future some mechanics described in #28227 could be added as well.

**Comment by Aivean** [comment]
Some relevant considerations:
* Instead of a bit matrix, current `fields` layer could be used. However, need to ensure that performance is still acceptable
* Snow would be actively deposited with a tile granularity in the "active map", but for non-loaded submaps a single float could be stored and updated according to the weather, so, when submap is loaded, the snow deposits are adjusted to the necessary level (that should be consistent among the neighboring submaps)

**Comment by Ramza13** [comment]
The field style of doing it is likely easier as weather already supports creating fields.  And setting something kinda generic up would be nice to allow stranger weathers like dust storms or possibly a way for strange temporary walls via magic spell or ray gun.

**Reply by Aivean** [reply]
Yes, fields would be a good fit, but I need to check if performance is acceptable.
